backend/src/database/state-migrations/migrations.ts

```diff
@@ -136,71 +128,86 @@ export async function migrateInDatabaseTo(
         }
         if (actionsToUpdate.length > 0) {
             await Promise.all(
-                actionsToUpdate.map(async ({ index, actionString }) =>
-                    entityManager.update(
-                        ActionWrapperEntity,
-                        { index, exercise: { id: exerciseId } },
-                        { actionString }
-                    )
+                actionsToUpdate.map(
+                    async ({ previousIndex, newIndex, actionString }) =>
+                        entityManager.update(
+                            ActionWrapperEntity,
+                            {
+                                index: previousIndex,
+                                exercise: { id: exerciseId },
+                            },
+                            { actionString, index: newIndex }
+                        )
                 )
             );
         }
     }
 }
 
-export function migrateStateExportTo(
-    targetStateVersion: number,
-    currentStateVersion: number,
-    stateExport: StateExport
-): StateExport {
-    const migrationFunctions = getMigrationFunctions(
-        currentStateVersion,
-        targetStateVersion
+export function migrateStateExport(stateExport: StateExport): StateExport {
+    const { currentState, history } = applyMigrations(
+        stateExport.dataVersion,
+        stateExport.currentState,
+        stateExport.history
+            ? {
+                  initialState: stateExport.history.initialState,
+                  actions: stateExport.history.actionHistory,
+              }
+            : undefined
     );
```

Comment:
`stateExport` is immutable. I believe this mutates `stateExport`.

Either change the input to `Mutable<StateExport>` or use `cloneDeep()`.